src/main/java/io/cdap/delta/bigquery/BigQueryAssessor.java

```diff
@@ -53,10 +62,58 @@ public TableAssessment assess(StandardizedTableDetail tableDetail) {
       }
     }
     List<Problem> problems = new ArrayList<>();
+    String dbName = tableDetail.getDatabase();
+    String tableName = tableDetail.getTable();
     if (tableDetail.getPrimaryKey().isEmpty()) {
-      problems.add(new Problem("Missing Primary Key", "Tables must have a primary key in order to be replicated.",
-                               "Please alter the table to use a primary key, or select a different table", ""));
+      problems.add(new Problem("Missing Primary Key",
+                               String.format("Table '%s' in database '%s' must have a primary key in order to be " +
+                                               "replicated", dbName, tableName),
+                               "Please alter the table to use a primary key, or select a different table",
+                               "Not able to replicate this table in BigQuery side"));
+    }
+
+    if (dbName.length() > MAX_LENGTH) {
+      problems.add(new Problem("Dataset Name Exceeding Max Length",
+                               String.format("Dataset name '%s' exceeds the max length %d defined in BigQuery",
+                                             dbName, MAX_LENGTH),
+                               "It's better to alter the database to make the name shorter",
+                               "There could be issues if multiple datasets would normalize to the same thing " +
+                                 "which will cause conflicts"));
+    }
+    if (!dbName.matches(VALID_NAME_REGEX)) {
```

Comment:
in most circumstances, users cannot rename their source databases and tables. Instead, the BQ target should normalize them automatically to something that BQ does support (for example, by substituting all invalid characters into underscore). Problems here will just highlight that they are being renamed, and that conflicts may arise.

So these problems and the name length can be combined into a single problem -- one that is really more of a warning but not necessarily an error.

Reply:
Got it. Makes sense for me. I will combine them together into single problem.